contrib/executor/jmeterd/pkg/parser/csvjtl.go

```diff
@@ -0,0 +1,80 @@
+package parser
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"strconv"
+	"time"
+)
+
+type CSVResults struct {
+	HasError         bool
+	LastErrorMessage string
+	Results          []CSVResult
+}
+
+type CSVResult struct {
+	Success      bool
+	Error        string
+	Label        string
+	ResponseCode string
+	Duration     time.Duration
+}
+
+func parseCSV(reader io.Reader) (results CSVResults, err error) {
+	res, err := CSVToMap(reader)
+	if err != nil {
+		return
+	}
+
+	for _, r := range res {
+		result := MapElementToResult(r)
+		results.Results = append(results.Results, result)
+
+		if !result.Success {
+			results.HasError = true
+			results.LastErrorMessage = result.Error
+		}
+	}
+
+	return
+}
+
+// CSVToMap takes a reader and returns an array of dictionaries, using the header row as the keys
+func CSVToMap(reader io.Reader) ([]map[string]string, error) {
+	r := csv.NewReader(reader)
+	var rows []map[string]string
+	var header []string
+	for {
+		record, err := r.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return nil, err
+		}
+		if header == nil {
+			header = record
+		} else {
+			dict := map[string]string{}
+			for i := range header {
+				dict[header[i]] = record[i]
+			}
+			rows = append(rows, dict)
+		}
+	}
+	return rows, nil
+}
+
+func MapElementToResult(in map[string]string) CSVResult {
+	elapsed, _ := strconv.Atoi(in["elapsed"])
```

Comment:
do we need to log error?

Reply:
Good question, we parse the JTL file here, it shouldn't ever error, unless JMeter changes it to something else in future versions which doesn't seem likely.
Even if it failed, I don't know should we fail the test, but might make sense to log the error at least